### PR TITLE
Fixes csharp build for 64 bit systems

### PR DESCRIPTION
**Pull request by bp-kelley**
swig is incredibly broken for 64 bit builds on windows, you can see the fixes in the commits, but basically 

(1) I had to properly define integers and 
(2) remove the INT_MAX==LONG_MAX check because, on windows, int == long.

How are the java 64 bit builds made?  I would assume that they have the same issue with stdint.i

### CODE REVIEW

**Comment by greglandrum** [comment]
The java 64bit builds seem to just work. I do them with MSVC 10 in order to maximize compatibility (the java wrapper binaries are used for the KNIME nodes and those are very broadly used), but I have a Win7 machine setup that I've been using for that for years.

**Comment by greglandrum** [comment]
Trying on my win10 laptop using a recent version of Visual Studio I can reproduce the problem in both the csharp and java wrappers.

**Comment by bp-kelley** [comment]
You may want to add the noexcept stuff from the java wrapper to the csharp at least.

I'm happy for now having a working build.

What version of swig are you using btw?
----
Brian Kelley

**Comment by greglandrum** [comment]
FYI: I'm doing a bit of tweaking of your code to get it working on my win10 laptop with both C# and java and then I'll do a PR against your repo.

I'm using swig 3.0.8

**Comment by bp-kelley** [comment]
Please do, this was more an informative "I can't believe this is necessary" pr.

Btw - I'm not happy with the Python requirement, we can probably do the same thing with a cmake script directly.

----
Brian Kelley

**Comment by greglandrum** [comment]
yeah, that's what I'm doing (the cmake-only version)

**Comment by greglandrum** [comment]
thanks for the PR though, finding and fixing this kind of thing is one of those real banging-your-head-on-the-table things

**Comment by bp-kelley** [comment]
I started having flashbacks to both CMake and SWIG nightmares.

File: Code/JavaWrappers/csharp_wrapper/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,8 @@ project (GraphMolCSharp)
 
 include_directories( ${RDKit_ExternalDir} )
 
+# requires python to fix broken swig .cxx files
```
Comment (greglandrum) [inline Code/JavaWrappers/csharp_wrapper/CMakeLists.txt]: This no longer needs to be here. Sorry I forgot to remove it in my PR

File: Code/JavaWrappers/gmwrapper/GraphMolJava.i [hunk 2]
```diff
@@ -49,7 +68,11 @@
 %include <boost/smart_ptr/shared_array.hpp>
 
 /* Include the base types before anything that will utilize them */
-%include "stdint.i"
+#ifdef SWIGWIN
```
Comment (greglandrum) [inline Code/JavaWrappers/gmwrapper/GraphMolJava.i]: This mod should also be made to the csharp wrappers. Those can be built on non-windows systems as well (using mono)

**Comment by greglandrum** [comment]
@bp-kelley I'm comfortable with merging this as soon as the tests pass. You agree?

**Comment by bp-kelley** [comment]
Comfortable is too strong a word :)  I'm ok with it though!

**Comment by greglandrum** [comment]
crossed my fingers and pressed the button! ;-)

**Pull request merged**